hpedockerplugin/hpe_storage_api.py

```diff
@@ -1,1196 +1,1259 @@
+                return json.dumps({u"Err": msg})
+
+            if snapname:
+                snapshots = vol['snapshots']
+                LOG.info("Getting snapshot by name: %s" % snapname)
+                snapshot, idx = self._get_snapshot_by_name(snapshots,
+                                                           snapname)
+
+                if snapshot:
+                    LOG.info("Found snapshot by name: %s" % snapname)
+                    # Does the snapshot have child snapshot(s)?
+                    for ss in snapshots:
+                        LOG.info("Checking if snapshot has children: %s"
+                                 % snapname)
+                        if ss['parent_id'] == snapshot['id']:
+                            msg = (_LE('snapshot %s/%s has one or more child '
+                                       'snapshots - it cannot be deleted!'
+                                       % (volname, snapname)))
+                            LOG.error(msg)
+                            # raise exception.HPEPluginRemoveException(
+                            # reason=msg)
+                            response = json.dumps({u"Err": msg})
+                            return response
+                    try:
+                        LOG.info("Deleting snapshot at backend: %s" % snapname)
+                        self.hpeplugin_driver.delete_volume(snapshot,
+                                                            is_snapshot=True)
+                    except Exception as err:
+                        msg = (_LE('Failed to remove snapshot error is: %s'),
+                               six.text_type(err))
+                        LOG.error(msg)
+                        response = json.dumps({u"Err": six.text_type(err)})
+                        try:
+                            self._etcd.try_unlock_volname(volname)
+                            lock_acquired = False
+                        except Exception as ex:
+                            LOG.error('volume: %(name)s Unlock Volume Failed',
+                                      {'name': volname})
+                            response = json.dumps({u"Err": six.text_type(ex)})
+                        return response
+
+                    LOG.info("Deleting snapshot in ETCD - %s" % snapname)
+                    # Remove snapshot entry from list and save it back to
+                    # ETCD DB
+                    del snapshots[idx]
+                    try:
+                        LOG.info("Updating volume in ETCD after snapshot "
+                                 "removal - vol-name: %s" % volname)
+                        # For now just track volume to uuid mapping internally
+                        # TODO: Save volume name and uuid mapping in etcd as
+                        # well. This will make get_vol_byname more efficient
+                        self._etcd.update_vol(vol['id'],
+                                              'snapshots',
+               
```

Comment:
Instead of calling driver to get latest data, we can populate the data from the etcd database.

The reason we are making a  query to an array get qos and snapshot recent data is : these two operations we are allowing to via docker (by setting expiration hours for snapshot and allowing predefined the qos rules, resulting in need to sync with an array).  Hence other properties like provisioning, compression or flash_cash we should populate directly from etcd only.

'vol_info' will have all the necessary details here which needs to be populated in 'volume_detail'

@wdurairaj @imran-ansari  can decide better on this